hail/src/test/scala/is/hail/expr/ir/IRSuite.scala

```diff
@@ -723,6 +723,21 @@ class IRSuite extends HailSuite {
     assertEvalsTo(If(True(), NA(TInt32()), I32(7)), null)
   }
 
+  @Test def testIfInferPType() {
+    assertPType(If(True(), I32(5), I32(7)), PInt32(true))
```

Comment:
the consequent and alternate IRs should be ptyped `In` nodes. Test the minimal thing possible -- don't rely on the ptype inference of other nodes like MakeArray.